File: 2.py
n = int(input())
routes = {}
starts = []
ends = []

for i in range(n):
    line = input().split()
    u, v = line[0], line[1]
    routes[u] = v
    starts.append(u)
    ends.append(v)

current = ""
for city in starts:
    if city not in ends:
        current = city
        break

while current in routes:
    current = routes[current]

print(current)

File: 2.exs
n_str = IO.gets("") |> String.trim()
n = String.to_integer(n_str)

tickets = Enum.map(1..n, fn _ ->
  [u, v] = IO.gets("") |> String.split()
  {u, v}
end)
routes = Enum.into(tickets, %{})
{starts, ends} = Enum.unzip(tickets)
start_city = Enum.find(starts, fn city -> city not in ends end)

step = fn func, city ->
  case Map.get(routes, city) do
    nil -> city
    next_city -> func.(func, next_city)
  end
end

IO.puts step.(step, start_city)

